### PR TITLE
feat: adding otlp_http trace exporter

### PR DESCRIPTION
**Pull request by varkey98**
## Description
Noticed it while creating the ecs template for TPA. Goagent as of now doesnt support otlp over http. Adding that support here.

### Testing
Tested both TLS and non TLS connection to TPA

### Checklist:
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,21 +24,22 @@ require (
...
 	go.opentelemetry.io/otel/exporters/stdout/stdouttrace v1.24.0
 	go.opentelemetry.io/otel/exporters/zipkin v1.24.0
 	go.opentelemetry.io/otel/metric v1.24.0
 	go.opentelemetry.io/otel/sdk v1.24.0
 	go.opentelemetry.io/otel/sdk/metric v1.24.0
 	go.opentelemetry.io/otel/trace v1.24.0
-	golang.org/x/net v0.23.0 // indirect
-	google.golang.org/grpc v1.62.1
+	golang.org/x/net v0.25.0 // indirect
+	google.golang.org/grpc v1.64.0
```
Comment (varkey98) [inline go.mod:36]: hopefully this wont cause any breaking issues

**Review by tim-mwangi — APPROVED** [review]
I wonder why codecov is having issues?

**Comment by varkey98** [comment]
> I wonder why codecov is having issues?

Its printing that it got rate limited. Maybe the licensing issue we face in our org as well?

**Review by ryanericson — APPROVED** [review]
Great!

**Comment by tim-mwangi** [comment]
> > I wonder why codecov is having issues?
> 
> Its printing that it got rate limited. Maybe the licensing issue we face in our org as well?

Can you ask in engineering channel? Maybe there's an answer.

**Review by senanjude0 — APPROVED** [review]

**Comment by varkey98** [comment]
> > > I wonder why codecov is having issues?
> > 
> > 
> > Its printing that it got rate limited. Maybe the licensing issue we face in our org as well?
> 
> Can you ask in engineering channel? Maybe there's an answer.

Upgrading the codecov version fixed it

**Pull request merged**